Makefile

```diff
@@ -58,8 +58,11 @@ lint-and-test: check-generate-init lint test-unit
 
 # ---- Integration Tests (algod required) ---- #
 
+sandbox-dev-up: SHELL:=/bin/bash
 sandbox-dev-up:
 	docker-compose up -d algod
+	SANDBOX_READY_REGEX="^Last committed block:*" ; \
+	while ! [[ `docker compose exec algod goal node status 2>&1 | head -n 1` =~ $$SANDBOX_READY_REGEX ]] ; do echo "waiting algod..." ; sleep 1 ; done
```

Comment:
not sure if writing shell in `Makefile` is the best practice, but this snippet of code serve as the guard for image to be prepared, before we start integration test.

Reply:
Could the `docker-compse up` command block until the algod node is up and running?

Reply:
We have to use detached mode, to run the container background, right?

Reply:
There is a `--wait` flag that will wait until the container is up and running and healthy and works in detached mode.

Might have to define healthchecks for it in the `docker-compose.yml` file that does something like we're doing here

Reply:
good point, I think we can make Makefile succinct, and move wait functionality somewhere else then

Reply:
@barnjamin take a pass? I did something in workflow to let `docker-compose` to work with `--with` for health check.

Reply:
ok we are moving from `docker-compose` to `docker compose`, activating command line arg `--wait` should be fine.